src/main/scala/org/apache/mesos/chronos/utils/JobSerializer.scala

```diff
@@ -180,33 +180,52 @@ class JobSerializer extends JsonSerializer[BaseJob] {
       case schedJob: ScheduleBasedJob =>
         json.writeFieldName("schedule")
         json.writeString(schedJob.schedule)
+        json.writeFieldName("scheduleType")
+        json.writeString(schedJob.scheduleType.name)
         json.writeFieldName("scheduleTimeZone")
         json.writeString(schedJob.scheduleTimeZone)
       case iSchedJob: InternalScheduleBasedJob =>
         json.writeFieldName("scheduleTimeZone")
         json.writeString(iSchedJob.scheduleTimeZone)
+
+        json.writeFieldName("scheduleType")
+        json.writeString(iSchedJob.scheduleData.scheduleType.name)
+
         json.writeFieldName("scheduleData")
         json.writeStartObject()
 
-        json.writeFieldName("schedule")
-        json.writeString(iSchedJob.scheduleData.schedule)
-        json.writeFieldName("scheduleTimeZone")
-        json.writeString(iSchedJob.scheduleData.scheduleTimeZone)
-        json.writeFieldName("originTime")
-        json.writeString(iSchedJob.scheduleData.originTime.toString(ISODateTimeFormat.dateTime))
-        json.writeFieldName("invocationTime")
-        json.writeString(iSchedJob.scheduleData.invocationTime.toString(ISODateTimeFormat.dateTime))
-        json.writeFieldName("offset")
-        json.writeNumber(iSchedJob.scheduleData.offset)
-
-        iSchedJob.scheduleData.recurrences.foreach { r =>
-          json.writeFieldName("recurrences")
-          json.writeNumber(r)
+        iSchedJob.scheduleData match {
+          case CronSchedule(schedule, scheduleTimeZone, invocationTime, lastExecutionTime, cron) =>
+            json.writeFieldName("schedule")
+            json.writeString(schedule)
+            json.writeFieldName("scheduleTimeZone")
+            json.writeString(scheduleTimeZone)
+            json.writeFieldName("invocationTime")
+            json.writeString(invocationTime.toString(ISODateTimeFormat.dateTime))
+            json.writeFieldName("lastExecutionTime")
+            json.writeString(lastExecutionTime.toString(ISODateTimeFormat.dateTime))
```

Comment:
"schedule", "scheduleTimeZone", and "invocationTime" could be written by a helper function shared between two schedule types.